lib/Sema/SemaBounds.cpp

```diff
@@ -3378,6 +3381,335 @@ namespace {
   };
 }
 
+namespace {
+class AvailableFactsAnalysis {
+private:
+  typedef llvm::SmallPtrSet<const Expr *, 16> ExprSet;
+  typedef std::pair<Expr *, Expr *> Comparison;
+  typedef std::set<Comparison> ComparisonSet;
+
+  class ElevatedCFGBlock {
+  private:
+    const CFGBlock *Block;
+    ComparisonSet In, OutThen, OutElse;
+    ComparisonSet Kill, GenThen, GenElse;
+
+  public:
+    ElevatedCFGBlock(const CFGBlock *Block) : Block(Block) {}
```

Comment:
@sunnychatterjee I followed a similar naming convention that appears in the existing constructors in this file. Should I change it?

Reply:
@ppashakhanloo You don't have to change this if it appears in the existing convention. Personally, I don't prefer this style since you have to qualify member access with a "this" in such cases.